File: src/commands/members.rs
use crate::client::AttioClient;
use crate::error::Result;
use crate::output::{print_many, print_one, OutputFormat};
use crate::types::{DataResponse, ListResponse, Member};

/// List workspace members.
pub async fn list(client: &AttioClient, json: bool) -> Result<()> {
    let response: ListResponse<Member> = client.get("/workspace_members").await?;
    print_many(&response.data, OutputFormat::from_json_flag(json))?;
    Ok(())
}

/// Get a specific workspace member.
pub async fn get(client: &AttioClient, member_id: &str, json: bool) -> Result<()> {
    let response: DataResponse<Member> = client
        .get(&format!("/workspace_members/{}", member_id))
        .await?;
    print_one(&response.data, OutputFormat::from_json_flag(json))?;
    Ok(())
}

File: src/attio_cli/commands/members.py
"""Workspace member commands."""

import click

from attio_cli.columns import MEMBER_COLUMNS
from attio_cli.commands.common import get_client
from attio_cli.output import output_many, output_one


@click.group()
def members() -> None:
    """Manage workspace members."""


@members.command("list")
@click.option("--json", "as_json", is_flag=True, help="Output as JSONL")
def members_list(as_json: bool) -> None:
    """List workspace members."""
    with get_client() as client:
        response = client.get("/workspace_members")
        output_many(response["data"], MEMBER_COLUMNS, as_json)


@members.command("retrieve")
@click.argument("member_id")
@click.option("--json", "as_json", is_flag=True, help="Output as JSON")
def members_retrieve(member_id: str, as_json: bool) -> None:
    """Retrieve member details."""
    with get_client() as client:
        response = client.get(f"/workspace_members/{member_id}")
        output_one(response["data"], MEMBER_COLUMNS, as_json)
